polygon/sync/execution_client.go

```diff
@@ -128,9 +135,11 @@ func (e *executionClient) CurrentHeader(ctx context.Context) (*types.Header, err
 	if err != nil {
 		return nil, err
 	}
-	if (response == nil) || (response.Header == nil) {
-		return nil, nil
+
+	if (response == nil) || (response.Header == nil) || response.Header.BlockNumber == 0 {
+		return e.blockReader.HeaderByNumber(ctx, nil, e.blockReader.FrozenBlocks())
```

Comment:
This feels like a hack. As I mentioned earlier I would prefer not to have a dependency on block reader in here at all.
This belongs elsewhere.
For example when the snapshot download is finished and the ExecutionEngine is Ready, this should already have been set by the ExecutionEngine

Reply:
This removed now the underlying start up issue is resolved